specs/tooltip.md

```diff
@@ -0,0 +1,74 @@
+# Tooltip
+
+## Overview
+
+A tooltip is a popup that displays information related to an element when the element receives keyboard focus or the mouse hovers over it. It typically appears after a small delay and disappears when Escape is pressed or on mouse out.
+
+### Use Cases
+
+Tooltips are often used to supplement UI elements and provide additional information. Tooltips will often provide accessible names or add additional context to other UI components (such as icon buttons). While tooltips are natively provided to elements via the title attribute, they are notoriously difficult to style. Additionally, they are only invoked on hover and not on focus.
+
+### Non-goals
+Tooltip widgets do not receive focus. A hover that contains focusable elements can be made using a non-modal dialog (flyout).
+
+### Prior Art/Examples
+- [Lightning Design](https://www.lightningdesignsystem.com/components/tooltips/)
+- [Office Fabric](https://developer.microsoft.com/en-us/fabric#/controls/web/tooltip)
+- [Evergreen](https://evergreen.segment.com/components/tooltip/)
+- [Carbon Design](https://www.carbondesignsystem.com/components/tooltip/code/)
+- [Ant Design](https://ant.design/components/tooltip/)
+- [Atlassian](https://atlaskit.atlassian.com/packages/core/tooltip)
+- [Bootstrap](https://getbootstrap.com/docs/4.3/components/tooltips/)
+
+---
+
+### API
+
+*Component name:*
+- `fast-tooltip`
+
+*Attributes:*
+- `trigger` - the html id of the HTMLElement which triggers the tooltip. This element references the tooltip element with aria-describedby.
+- `delay` - time in milliseconds to wait before showing and hiding the tooltip. Defaults to 300.
+- `visible` - the visiblity of the tooltip
+- `position` - enum; where the tooltip should appear relative to its target.
+    - bottom
+    - left
+    - right
+    - top
+
+*Slots:*
+- `default`
+
+*Events*
+- dismiss - event fired when a user presses escape
+
+### Anatomy and Appearance
+
+*Template:*
+```
+<template
+    role="tooltip"
+>
+    <slot></slot>
+</template>
+```
+
+## Implementation
+
+```
+<button id="foo" aria-describedby="tooltip">Foo</button>
```

Comment:
That's such a bummer that these need to cross-reference each other - I wish there was an `aria-describes` attribute.

Reply:
Same - I wonder if anything like that has ever been floated?

Reply:
Could we do this automatically in the tooltip? So, when the tooltip's associated control changes, if that control doesn't have the described-by attribute, we set it up. (we can generate a unique id if there isn't an id set as well)

Reply:
@EisenbergEffect it seems reasonable that we could use the reference provided to check for the existence of the attribute and then assign it if does not exist.

Reply:
I think the scenario where you have a single tooltip instance and change it's anchor dynamically would be fairly common, so tooltip should be able to attach/detach from an anchor element.

Reply:
Thanks @scomea, I think that would simply be handled by a changed event when the anchor or what not changes which seems implementation heavy to include in the spec - but I agree with the premise outlined above.